keras_cv/models/resnet_v2_backbone.py

```diff
@@ -482,189 +377,20 @@ def get_config(self):
-        stackwise_blocks=MODEL_CONFIGS["ResNet50V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet50V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=parse_weights(weights, include_top, "resnet50v2"),
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
-        **kwargs,
-    )
-
-
-def ResNet101V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet101v2",
-    **kwargs,
-):
-    """Instantiates the ResNet101V2 architecture."""
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet101V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet101V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet101V2"]["stackwise_strides"],
-        name=name,
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
-        **kwargs,
-    )
-
-
-def ResNet152V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet152v2",
-    **kwargs,
-):
-    """Instantiates the ResNet152V2 architecture."""
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet152V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet152V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet152V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
-        **kwargs,
-    )
-
+    @classproperty
+    def presets(cls):
+        return copy.deepcopy(backbone_presets)
 
-setattr(ResNet18V2, "__doc__", BASE_DOCSTRING.format(name="ResNet18V2"))
-setattr(ResNet34V2, "__doc__", BASE_DOCSTRING.format(name="ResNet34V2"))
-setattr(ResNet50V2, "__doc__", BASE_DOCSTRING.format(name="ResNet50V2"))
-setattr(ResNet101V2, "__doc__", BASE_DOCSTRING.format(name="ResNet101V2"))
-setattr(ResNet152V2, "__doc__", BASE_DOCSTRING.format(name="ResNet152V2"))
+    @classproperty
+    def presets_with_weights(cls):
```

Comment:
I like that theres an abstraction here to make this easier/more browsable.

Reply:
@fchollet had an idea to improve this further by diverging from the KerasNLP API. Instead of a `load_weights` argument we could have some presets come without weights (`"resnetv2_50"`) and other with weights (`"resnetv2_50_imagenet"`). Our documentation / docstrings would have to clarify which was which.

Reply:
It is usually the case in CV that architectures and their weights are entirely orthogonal. However it's nice that we have an API that is general enough to support both situations -- the orthogonal case and the interdependent case. If we run into architectures that are checkpoint-dependent it's easy to add a preset for them. Should we keep the `load_weights` arg or not?

Reply:
I used the `None` default trick to keep this arg in the signature.